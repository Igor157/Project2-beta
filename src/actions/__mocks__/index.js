

// const action = jest.genMockFromModule('../index');

export function getCur() {

    return (dispatch) => {
        dispatch({
            type: 'GET_CUR_REQUEST'
        });
        dispatch({
            type: 'GET_CUR_SUCCESS',
            payload: [{
                "Cur_ID": 170,
                "Date": "2018-02-20T00:00:00",
                "Cur_Abbreviation": "AUD",
                "Cur_Scale": 1,
                "Cur_Name": "Австралийский доллар",
                "Cur_OfficialRate": 1.5533
            },
            {
                "Cur_ID": 191,
                "Date": "2018-02-20T00:00:00",
                "Cur_Abbreviation": "BGN",
                "Cur_Scale": 1,
                "Cur_Name": "Болгарский лев",
                "Cur_OfficialRate": 1.2443
            }]
        });
    };
}

export function filterAvailableCur(value) {
    return {
        type: 'FILTER_AVAILABLE_CUR',
        payload: {
            filterText: value
        }
    };
}

export function changeCurForDynamic(id, abr) {
    return {
        type: 'CHANGE_CUR_FOR_DYNAMIC',
        payload: {
            changedId: id,
            changedAbr: abr
        }
    };
}

export function getDynamic(curID, start, end) {

    return (dispatch) => {
        dispatch({
            type: 'GET_DYNAMIC_REQUEST'
        });
        dispatch({
            type: 'GET_DYNAMIC_SUCCESS',
            payload: 'someData'
        });
    };
}