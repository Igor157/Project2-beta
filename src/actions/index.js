import { requestServices } from '../Services/Services.js';
import { converterServices } from '../Services/Services.js';


export const filterCur = (value) => {
    return {
        type: 'FILTER_CUR',
        payload: value
    };
};

export const chooseCur = (id, abr) => {
    return {
        type: 'CHOOSE_CUR',
        payload: {
            id: id,
            abr: abr
        }
    };
};
// export const initialize = () => {
//     return {
//         type: 'INITIALIZE',
//         payload: {
//             cur: [],
//             filterText: false
//         }
//     }
// }


export function getCur() {

    return (dispatch) => {
        dispatch({
            type: 'GET_CUR_REQUEST'
        });
        let todaysCurPromise = requestServices.getTodaysCurrencies();
        todaysCurPromise
            .then(data => {
                dispatch({
                    type: 'GET_CUR_SUCCESS',
                    payload: data
                });
            });
    };
}

export function getDynamic(curID, start, end) {

    return (dispatch) => {
        dispatch({
            type: 'GET_DYNAMIC_REQUEST'
        });
        let dynamicForIdPromise = requestServices.getDynamicForCurId(curID, start, end);
        dynamicForIdPromise
            .then(data => {
                dispatch({
                    type: 'GET_DYNAMIC_SUCCESS',
                    payload: data
                });
            });
    };
}
export function changeCurForDynamic(id, abr) {
    return {
        type: 'CHANGE_CUR_FOR_DYNAMIC',
        payload: {
            changedId: id,
            changedAbr: abr
        }
    };
}
export function filterAvailableCur(value) {
    return {
        type: 'FILTER_AVAILABLE-CUR',
        payload: {
            filterText: value
        }
    };
}

export function changeValue1(target) {
    return {
        type: 'CHANGE_VALUE_1',
        payload: {
            inputValue: target
        }
    };
}
export function changeValue2(target) {
    return {
        type: 'CHANGE_VALUE_2',
        payload: {
            inputValue: target
        }
    };
}
export function changeAbr1(target, cur) {
    return {
        type: 'CHANGE_ABR_1',
        payload: {
            input1Abr: target,
            current1Rate: converterServices.filterCurForTarget(cur, target) ?
                converterServices.filterCurForTarget(cur, target).curRate
                : ''
        }
    };
}
export function changeAbr2(target, cur) {
    return {
        type: 'CHANGE_ABR_2',
        payload: {
            input2Abr: target,
            current2Rate: converterServices.filterCurForTarget(cur, target) ?
                converterServices.filterCurForTarget(cur, target).curRate
                : ''
        }
    };
}

export function changeStartDate(date) {
    return {
        type: 'CHANGE_START_DATE',
        payload: {
            startDate: date
        }
    };
}
export function changeEndDate(date) {
    return {
        type: 'CHANGE_END_DATE',
        payload: {
            endDate: date
        }
    };
}
export function addCurToFavorite(abr, id, dynamic, favoriteFirst) {
    return {
        type: 'ADD_CUR_TO_FAVORITE',
        payload: {
            favoriteId: id,
            favoriteAbr: abr,
            dynamic: dynamic,
            selected: favoriteFirst
        }
    };

}
export function selectFavoriteCur(targetId) {
    return {
        type: 'SELECT_FAVORITE_CUR',
        payload: {
            id: targetId,
            new: false,
            selected: true
        }
    };

}
export function removeFromFavorite(targetId) {
    return {
        type: 'REMOVE_FROM_FAVORITE',
        payload: {
            id:targetId
        }
    };
}