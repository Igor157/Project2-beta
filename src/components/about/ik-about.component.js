import React from 'react';
import ReactDOM from 'react-dom';
import styles from './ik-about.style.css';


export class About extends React.Component {
  render() {
    const pageElementClass = this.props.className;
    return (
      <div className={`ik-about ${pageElementClass}`} >
        <div className="ik-about__author">This site was designed and developed by: Igor Kazakov</div>
        <div className="ik-about__link">GitHub URL:
          <a href="https://github.com/Igor157/FrontEnd-Project-2">
            https://github.com/Igor157/FrontEnd-Project-2
          </a>
        </div>
        <div className="ik-about__link">GitHub pages:
          <a href="https://igor157.github.io/Currencies-project/#/">
            https://github.com/Igor157/FrontEnd-Project-2
          </a>
        </div>
        <div className="ik-about__link">Design:
          <a href="https://github.com/Igor157/Project2-beta/blob/master/files%20for%20download/Currency.pdf" download>
            Currency.pdf
          </a>
        </div>
      </div>
    );
  }
}