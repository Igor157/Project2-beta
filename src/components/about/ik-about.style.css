.ik-about {
    color: rgb(74, 74, 228);
    font-size: 1.6em;
}

.ik-about__author {}

.ik-about__link {}