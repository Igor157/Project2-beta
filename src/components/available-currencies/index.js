export * from './ik-available-currencies.component.js';