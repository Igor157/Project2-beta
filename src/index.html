<!doctype html>
<html lang="ru">

<head>
  <script src="https://use.fontawesome.com/fd0afdd84c.js"></script>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width">
  <title>
    <%= htmlWebpackPlugin.options.title %>
  </title>
</head>

<body>
  <div class='ik-page' id='ik-page'></div>
</body>

</html>