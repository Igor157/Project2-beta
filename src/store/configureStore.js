import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../reducers';
import createLogger from 'redux-logger';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import {todoApp} from '../reducers/index.js';
import persistState from 'redux-localstorage';

export default function configureStore(initialState) {
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    todoApp,
    initialState,
    composeEnhancers(
    applyMiddleware(thunk, logger),
    persistState()
  ));
  return store;
}
