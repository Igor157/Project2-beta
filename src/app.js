import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { CurrencyPage } from './components/currency-page';
import configureStore from './store/configureStore.js';
import moment from 'moment';

import {
    HashRouter as Router,
    Route,
    NavLink,
    Switch
} from 'react-router-dom';
import { createStore } from 'redux';
let initDateForStat = { startDate: moment().date(-20), endDate: moment() };
let store = configureStore({
    getCurrencies: { cur: [] },
    getDynamic: { dynamic: [] },
    curToFavorite: {
        empty: true,
        favoriteCurData: []
    },
    changeDate: {
        startDate: initDateForStat.startDate,
        endDate: initDateForStat.endDate
    },
    getInfo: {
        info: []
    }
});
ReactDOM.render(
    <Provider store={store}>
        <Router>
            <CurrencyPage />
        </Router>
    </Provider>,
    document.getElementById('ik-page')
);

