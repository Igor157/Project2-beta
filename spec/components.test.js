import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import * as actions from '../src/actions';
import { Search } from '../src/components/search';
import { FilterSearch } from '../src/containers/ik-filter-search.container';
import { About } from '../src/components/about';
import { AvailableCurrencies } from '../src/components/available-currencies';
import { ChooseAvailableCurrencies } from '../src/containers/ik-choose-available-currencies.container';
// import { requestServices } from '../src/services/Services.js';

import thunk from 'redux-thunk';
const middlewares = [thunk];
configure({ adapter: new Adapter() });



test('Snapshot test of Search', () => {
    const component = renderer.create(
        <Search filterText='USD' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

describe('FilterSearch container should map states and dispatch actions', () => {
    const initialState = { availableCurrencies: { filterText: 'USD' } };
    const mockStore = configureStore(middlewares);
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><FilterSearch /></Provider>);
    });


    test('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(FilterSearch).length).toEqual(1);
    });

    test('+++ check Prop matches with initialState', () => {
        expect(wrapper.find(Search).prop('filterText')).toEqual(initialState.availableCurrencies.filterText);
    });

    test('+++ check action on dispatching ', () => {
        wrapper.find('.ik-search__search-place').simulate('change');
        let action = store.getActions();
        expect(action[0].type).toBe('FILTER_AVAILABLE_CUR');
    });

});

test('Snapshot test of About', () => {
    const component = renderer.create(
        <About />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

let mockCur = [{
    curAbr: "AUD",
    curRate: 1.5285,
    date: "March 8th 2018",
    curId: 170,
    curName: "",
    startDate: "",
    endDate: "",
    curScale: 1,
    curDifference: 0.0030999999999998806
},
{
    curAbr: "BGN",
    curRate: 1.2443,
    date: "March 8th 2018",
    curId: 191,
    curName: "",
    startDate: "",
    endDate: "",
    curScale: 1,
    curDifference: 0.0043999999999999595
},
{
    curAbr: "UAH",
    curRate: 7.5032,
    date: "March 8th 2018",
    curId: 290,
    curName: "",
    startDate: "",
    endDate: "",
    curScale: 100,
    curDifference: 0.036399999999999544
}];

test('Snapshot test of AvailableCurrencies', () => {
    jest.mock('../src/actions/index');
    const component = renderer.create(
        <AvailableCurrencies
            className='ik-currency-page__available-currencies'
            cur={mockCur}
            getCur={jest.fn()}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

jest.mock('../src/actions/index');

describe('ChooseAvailableCurrencies container should map states and dispatch actions', () => {
    const initialState = {
        getCurrencies: {
            cur: mockCur
        },
        availableCurrencies: {
            filterText: '',
            choosenId: '145'
        }
    };
    const mockStore = configureStore(middlewares);
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}>
            <ChooseAvailableCurrencies
                className='ik-currency-page__available-currencies'
            />
        </Provider>);
    });
    test('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(ChooseAvailableCurrencies).length).toEqual(1);
    });

    it('+++ check Prop matches with initialState', () => {
        expect(wrapper.find(AvailableCurrencies).prop('filterText')).toEqual(initialState.availableCurrencies.filterText);
        expect(wrapper.find(AvailableCurrencies).prop('cur')).toEqual(initialState.getCurrencies.cur);
        expect(wrapper.find(AvailableCurrencies).prop('choosenId')).toEqual(initialState.availableCurrencies.choosenId);
    });

    it('+++ check action on dispatching ', () => {
        console.log(wrapper.find('.ik-available-currencies__row').at(0));
        wrapper.find('.ik-available-currencies__row').at(0).simulate('click');
        let action = store.getActions();
        console.log(action);
        expect(action[0].type).toBe('GET_CUR_REQUEST');
        expect(action[1].type).toBe('GET_CUR_SUCCESS');
        expect(action[2].type).toBe('CHANGE_CUR_FOR_DYNAMIC');
    });

});
