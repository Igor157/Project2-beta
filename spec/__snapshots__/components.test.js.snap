// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of About 1`] = `
<div
  className="ik-about undefined"
>
  <div
    className="ik-about__author"
  >
    This site was designed and developed by: Igor Kazakov
  </div>
  <div
    className="ik-about__link"
  >
    GitHub URL:
    <a
      href="https://github.com/Igor157/FrontEnd-Project-2"
    >
      https://github.com/Igor157/FrontEnd-Project-2
    </a>
  </div>
  <div
    className="ik-about__link"
  >
    GitHub pages:
    <a
      href="https://igor157.github.io/ik-currency-project/#/"
    >
      https://igor157.github.io/ik-currency-project/#/
    </a>
  </div>
  <div
    className="ik-about__link"
  >
    Design:
    <a
      download={true}
      href="https://github.com/Igor157/Project2-beta/blob/master/files%20for%20download/Currency.pdf"
    >
      Currency.pdf
    </a>
  </div>
</div>
`;

exports[`Snapshot test of AvailableCurrencies 1`] = `
<div
  className="ik-available-currencies  ik-currency-page__available-currencies"
  onClick={[Function]}
>
  <div
    abr="AUD"
    className="ik-available-currencies__row"
    id={170}
    smth="ik-currency-tab"
  >
    <span
      aria-hidden={true}
      className="fa fa-money ik-available-currencies__icon"
      style={
        Object {
          "color": "white",
        }
      }
    />
    <div
      className="ik-available-currencies__abr"
    >
      AUD
    </div>
    <div
      className="ik-available-currencies__rate"
    >
      1.52850
    </div>
    <button
      className="ik-available-currencies__day-progress  ik-available-currencies__day-progress--green"
    >
      +0.0031
    </button>
  </div>
  <div
    abr="BGN"
    className="ik-available-currencies__row"
    id={191}
    smth="ik-currency-tab"
  >
    <span
      aria-hidden={true}
      className="fa fa-money ik-available-currencies__icon"
      style={
        Object {
          "color": "white",
        }
      }
    />
    <div
      className="ik-available-currencies__abr"
    >
      BGN
    </div>
    <div
      className="ik-available-currencies__rate"
    >
      1.24430
    </div>
    <button
      className="ik-available-currencies__day-progress  ik-available-currencies__day-progress--green"
    >
      +0.0044
    </button>
  </div>
  <div
    abr="UAH"
    className="ik-available-currencies__row"
    id={290}
    smth="ik-currency-tab"
  >
    <span
      aria-hidden={true}
      className="fa fa-money ik-available-currencies__icon"
      style={
        Object {
          "color": "white",
        }
      }
    />
    <div
      className="ik-available-currencies__abr"
    >
      UAH
    </div>
    <div
      className="ik-available-currencies__rate"
    >
      0.07503
    </div>
    <button
      className="ik-available-currencies__day-progress  ik-available-currencies__day-progress--green"
    >
      +0.0364
    </button>
  </div>
</div>
`;

exports[`Snapshot test of Search 1`] = `
<div
  className="ik-search undefined"
>
  <form
    className="ik-search__form"
  >
    <input
      className="ik-search__search-place"
      onChange={[Function]}
      placeholder="Search input"
      type="text"
      value="USD"
    />
    <div
      className="ik-search__search-button"
    >
      Search
    </div>
  </form>
</div>
`;
